contracts/contracts/validator-registry/ValidatorRegistryV1.sol

```diff
@@ -174,39 +183,55 @@ contract ValidatorRegistryV1 is IValidatorRegistryV1, ValidatorRegistryV1Storage
         _setUnstakePeriodBlocks(newUnstakePeriodBlocks);
     }
 
+    /// @dev Internal function that splits msg.value stake to apply an action for each validator.
+    function _splitStakeAndApplyAction(
+        bytes[] calldata blsPubKeys,
+        address withdrawalAddress,
+        function(bytes calldata, uint256, address) internal action
+    ) internal {
+        require(blsPubKeys.length != 0, "At least one recipient required");
+        uint256 baseStakeAmount = msg.value / blsPubKeys.length;
+        require(baseStakeAmount != 0, "Stake too low for number of keys");
+        uint256 lastStakeAmount = msg.value - (baseStakeAmount * (blsPubKeys.length - 1));
+        for (uint256 i = 0; i < blsPubKeys.length; ++i) {
```

Comment:
```suggestion
        uint256 numKeys = blsPubKeys.length;
        for (uint256 i = 0; i < numKeys; ++i) {
```
I would recommend to store the length of blsPubKeys in a variable to avoid repeated .length lookups